### PR TITLE
adds client api integration tests for runc using bash w/bats

### PR DESCRIPTION
**Pull request by mikebrow**
Closes issue #485

See readme.md for usage instructions.

Signed-off-by: Mike Brown <brownwm@us.ibm.com>

### CODE REVIEW

**Comment by marcosnils** [comment]
@mikebrow simply amazing!.

**Comment by hqhq** [comment]
Looks great! Thanks @mikebrow

**Comment by mikebrow** [comment]
Fix for the broken test is in #669

**Comment by cyphar** [comment]
Looks great. I reckon we should also add some tests making sure that resource limiting *actually* works (it was broken a few times in the past and nobody noticed until someone tried to vendor it into Docker).

**Comment by mikebrow** [comment]
Ayup.. next on my list was actually going to be stress & bootstrap tests.. like constraint checking.  First one I started writing I found a bug :-)

**Comment by mikebrow** [comment]
rebased on current master...

**Comment by crosbymichael** [comment]
LGTM

**Comment by cyphar** [comment]
LGTM. We should add this to the Jenkins test jobs.

File: script/test_Dockerfile [hunk 1]
```diff
@@ -16,16 +16,24 @@ RUN apt-get update && apt-get install -y \
     python-minimal \
     --no-install-recommends
 
+# install bats 
+RUN cd /tmp \
```
Comment (cyphar) [inline script/test_Dockerfile]: We should specify an exact commit or version in the repo to checkout. This means that we don't have to worry about inconsistencies in the test framework.

**Comment by crosbymichael** [comment]
ya, we can add that in another pr

File: script/test_Dockerfile (hunk 1 above)
Comment (mikebrow) [inline script/test_Dockerfile]: removed the trailing whitespace from this file... Per Michael's comment leaving the which/how of picking or caching a particular version of bats for another PR.

**Comment by mikebrow** [comment]
Rebased... comment addressed.

File: script/test_Dockerfile (hunk 1 above)
Comment (cyphar) [inline script/test_Dockerfile]: I think he meant "we should add it to the Jenkins jobs in another PR". I don't like pulling in the latest version of bats from HEAD when doing testing -- that doesn't strike me as being a particularly good idea.

Comment (mikebrow) [inline script/test_Dockerfile]: kk... np... updated to reset head back to a known commit (which is the latest atm).

Cheers.

**Comment by mikebrow** [comment]
Comments addressed.

**Comment by cyphar** [comment]
LGTM.

**Pull request merged**